database/models.go

```diff
@@ -29,34 +29,47 @@ type Processors struct {
 // Ancestry is a manifest that keeps all layers in an image in order.
 type Ancestry struct {
 	Name string
+	// ProcessedBy contains the processors that are used when computing the
+	// content of this ancestry.
+	ProcessedBy Processors
 	// Layers should be ordered and i_th layer is the parent of i+1_th layer in
 	// the slice.
 	Layers []Layer
 }
 
-// AncestryWithFeatures is an ancestry with namespaced features detected in the
-// ancestry, which is processed by `ProcessedBy`.
-type AncestryWithFeatures struct {
+// AncestryWithContent has the ancestry's name and the Ancestry Layers
+// associated with it.
+type AncestryWithContent struct {
 	Ancestry
 
-	ProcessedBy Processors
-	Features    []NamespacedFeature
+	// TODO(sidchen) deduplicate the Layers.
```

Comment:
is this something we want to do? won't the different layers have different context?

Reply:
Well, the problem here is that Ancestry has `Layers` field, which contains a list of `Layer`, and AncestryLayer also has `Layer`. The information is duplicated. I just don't really know a good way to deduplicate these.

Reply:
Why not have both types be the same but leave the DetectedFeatures for each Layer empty when it isn't AncestryWithContent?

Reply:
I remember the design is that: Every struct, when returned, MUST have all fields filled to represent the whole information. Therefore, leaving DetectedFeatures empty will indicate that the Layer has NO detected features.